### PR TITLE
Migrating work from Xins branch into a more up to date branch.

### PR DESCRIPTION
**Pull request by WAftring**

### CODE REVIEW

**Comment by WAftring** [comment]
Per discussion @to-whitney to work on additional test coverage.

**Review by hanwenzhang123 — APPROVED** [review]

File: code/SoloSavingsApp/bin/init-database.sql
Comment (AWOuellette) [inline code/SoloSavingsApp/bin/init-database.sql:1]: This should not have been added.

Comment (to-whitney) [reply code/SoloSavingsApp/bin/init-database.sql:1]: did i need to remove this even if its in bin folder?

Comment (to-whitney) [reply code/SoloSavingsApp/bin/init-database.sql:1]: This is resolved by not including the bin folder in the pr

**Comment by to-whitney** [comment]
I'm done with tests, coverage is at ~80% now and i think the tests cover most of the business logic so i think we are good. @WAftring

**Pull request merged**